### ISSUE TITLE
Path Parameters

### ISSUE THREAD

**Issue by constantine2nd** [OpenBankProject/OBP-API#1079]
Path parameters are variable parts of a URL path. They are typically used to point to a specific resource within a collection, such as a user identified by ID. 
For instance: **/users/{userId}**

OBP-API uses next notation:  **/users/USER_ID**

We need to consider using curly braces i.e **{userId}** instead of **USER_ID** in documentation.

**Comment by constantine2nd** [comment]
Related issues: #692, #693, #694, #696, #697

**Comment by chrisjsimpson** [comment]
Yes. The motivation here is clarity of variable parts of the api call vs paths.
It's pretty standard convention by now to refer to variable parts with brackets. Some examples:

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#binding-v1-core

![image](https://user-images.githubusercontent.com/1718624/57783672-719ba880-7726-11e9-995e-ff8b27a2a302.png)

Giant swarm: https://docs.giantswarm.io/api/#operation/createUser
![image](https://user-images.githubusercontent.com/1718624/57783773-a27bdd80-7726-11e9-9f10-11e4128485d6.png)

Without this it can be confusing for developers seeing the api for the first time. e.g. for Open Bank Project's [Grant User Access View](https://apiexplorer.openbankproject.com/?ignoredefcat=true&tags=#OBPv1_2_1-addPermissionForUserForBankAccountForOneView):

The api call is confusing because it's not obvious which parts the developer may change. 
`/obp/v3.1.0/banks/BANK_ID/accounts/ACCOUNT_ID/permissions/PROVIDER/PROVIDER_ID/views/VIEW_ID`

Compared to much clearer:
`/obp/v3.1.0/banks/{bankId}/accounts/{accountId}/permissions/{provider}/{providerId}/views/{viewId}`

> There should be one-- and preferably only one --obvious way to do it. Although that way may not be obvious at first unless you're Dutch. :wink: 



Implementing this improvement would impact:

- API Explorer 
- Documentation
- OpenAPI Generation 
At the benefit of :chart_with_upwards_trend: developer experience  :smile: 

@simonredfern
